File: programmers/js_src/level01/68644.js
// 두 개 뽑아서 더하기
function solution(numbers) {
  var answer = [];

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      answer.push(numbers[i] + numbers[j]);
    }
  }

  answer = Array.from(new Set(answer)).sort((a, b) => a - b);

  return answer;
}

// 다른 풀이
// function solution(numbers) {
//   const temp = []

//   for (let i = 0; i < numbers.length; i++) {
//     for (let j = i + 1; j < numbers.length; j++) {
//       temp.push(numbers[i] + numbers[j])
//     }
//   }

//   const answer = [...new Set(temp)]

//   return answer.sort((a, b) => a - b)
// }

File: programmers/java/level01/68644.java
// 두 개 뽑아서 더하기

import java.util.ArrayList;

class Solution {
    public int[] solution(int[] numbers) {
        ArrayList<Integer> result = new ArrayList<>();
        
        for (int i = 0; i < numbers.length - 1; i++) {
            for (int j = i + 1; j < numbers.length; j++) {
                result.add(numbers[i] + numbers[j]);
            }
        }
        
        return result.stream()
            .distinct()
            .sorted()
            .mapToInt(v -> v)
            .toArray();
    }
}